awscli/topics/config-vars.rst

```diff
@@ -0,0 +1,209 @@
+
+=========== ========= ===================== ===================== ============================
+Variable    Option    Config Entry          Environment Variable  Description
+=========== ========= ===================== ===================== ============================
+profile     --profile profile               AWS_DEFAULT_PROFILE   Default profile name
+----------- --------- --------------------- --------------------- ----------------------------
+region      --region  region                AWS_DEFAULT_REGION    Default AWS Region
+----------- --------- --------------------- --------------------- ----------------------------
+output      --output  output                AWS_DEFAULT_OUTPUT    Default output style
+=========== ========= ===================== ===================== ============================
+
+The third column, Config Entry, is the value you would specify in the AWS CLI
+config file.  By default, this location is ``~/.aws/config``.  If you need to
+change this value, you can specify the ``AWS_CONFIG_FILE`` environment variable
+to change this location.
+
+The valid values of the ``output`` configuration variable are:
+
+* json
+* table
+* text
+
+
+Credentials
+===========
+
+Credentials can be specified in several ways:
+
+* Environment variables
+* The AWS Shared Credential File
+* The AWS CLI config file
+
+=========== ===================== ===================== ============================
+Variable    Creds/Config Entry    Environment Variable  Description
+=========== ===================== ===================== ============================
+access_key  aws_access_key_id     AWS_ACCESS_KEY_ID     AWS Access Key
+----------- --------------------- --------------------- ----------------------------
+secret_key  aws_secret_access_key AWS_SECRET_ACCESS_KEY AWS Secret Key
+----------- --------------------- --------------------- ----------------------------
+token       aws_session_token     AWS_SESSION_TOKEN     AWS Token (temp credentials)
+=========== ===================== ===================== ============================
+
+The second column specifies the name that you can specify in either the AWS CLI
+config file or the AWS Shared credentials file (``~/.aws/credentials``).
+
+
+The Shared Credentials File
+---------------------------
+
+The shared credentials file has a fixed location of
+``~/.aws/credentials``.  This file is an INI formatted file with section names
+corresponding to profiles.  With each section, the three configuration
+variables shown above can be specified: ``aws_access_key_id``,
+``aws_secret_access_key``, ``aws_session_token``.  **These are the only
+supported values in the shared credential file.**
+
+Precedence
+----------
+
+Credentials from environment variables have precedence over credentials from
+the shared credentials and AWS CLI config file.  Credentials specified in the
+shared credentials file have precedence over credentials in the AWS CLI config
+file.
```

Comment:
How about precedence for region and values specified on the command line? This could get real wordy. You may want to break this up to a nice diagram or table to make it really easy to tell precedence.